sdk/src/androidTest/java/com/mapbox/maps/OfflineTest.kt

```diff
@@ -0,0 +1,627 @@
+        if (expected.isValue) {
+          expected.value?.let { stylePackList = it }
+        }
+        if (expected.isError) {
+          expected.error?.let { stylePackError = it }
+        }
+        latch.countDown()
+      }
+    }
+    if (!latch.await(10, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    }
+    return Pair(stylePackList, stylePackError)
+  }
+
+  private fun queryTileRegions(): Pair<List<TileRegion>, TileRegionError?> {
+    var tileRegionList: List<TileRegion> = listOf()
+    var tileRegionError: TileRegionError? = null
+    val latch = CountDownLatch(1)
+    TileStore.getInstance().getAllTileRegions { expected ->
+      if (expected.isValue) {
+        expected.value?.let { tileRegionList = it }
+      }
+      if (expected.isError) {
+        expected.error?.let { tileRegionError = it }
+      }
+      latch.countDown()
+    }
+    if (!latch.await(10, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    }
+    return Pair(tileRegionList, tileRegionError)
+  }
+
+  private fun disableAmbientCache() {
+    val latch = CountDownLatch(1)
+    // set 0 size ambient cache to make use of only tile store + style packs
+    handler.post {
+      cacheManager.setMaximumAmbientCacheSize(0) {
+        if (it.isValue) {
+          latch.countDown()
+        }
+      }
+    }
+    if (!latch.await(5, TimeUnit.SECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  private fun loadTileStoreWithStylePack() {
+    val latch = CountDownLatch(1)
+    handler.post {
+      TileStore.getInstance().loadTileRegion(
+        TILE_REGION_ID,
+        TileRegionLoadOptions.Builder()
+          .geometry(TOKYO)
+          .descriptors(
+            listOf(
+              offlineManager.createTilesetDescriptor(
+                TilesetDescriptorOptions.Builder()
+                  // load style pack here and not separately to reduce amount of code
+                  .styleURI(STYLE)
+                  .stylePackOptions(
+                    StylePackLoadOptions.Builder()
+                      .glyphsRasterizationMode(GlyphsRasterizationMode.IDEOGRAPHS_RASTERIZED_LOCALLY)
+                      .metadata(Value(STYLE_PACK_METADATA))
+                      .build()
+                  )
+                  // set min zoom twice less in order to make sure tile in loaded map view
+                  // will be fully visible and not somewhere on tile edge
+                  .minZoom((ZOOM.toInt() / 2).toByte())
+                  .maxZoom(ZOOM.toInt().toByte())
+                  .build()
+              )
+            )
+          )
+          .metadata(Value(TILE_REGION_METADATA))
+          .tileLoadOptions(
+            TileLoadOptions.Builder()
+              .networkRestriction(NetworkRestriction.NONE)
+              .criticalPriority(false)
+              .acceptExpired(true)
+              .build()
+          )
+          .build(),
+        {}
+      ) {
+        if (it.isValue) {
+          it.value?.let { latch.countDown() }
```

Comment:
This callback will only report the loading finished status for tile region, it will not reflect the status of the style pack, so  potentially the style pack can be still downloading at this point of time. And that's also the reason we were downloading the style pack and tile region separately in the OfflineActivity.

Reply:
@pozdnyakov can you please confirm that behaviour: `TileRegionCallback` could get invoked before style pack as part of descriptor gets downloaded. For me it seems more like a bug - I would expect `TileRegionCallback` be triggered after style pack being loaded as part of tile options descriptor.

Reply:
if the tileset descriptor is configured so that it creates a style pack, the style pack is created by the time the tileset descriptor gets resolved i.e. before the tile region loading starts

Reply:
@pozdnyakov thanks for confirming!
@pengdev - it seems that we're OK here morevover it looks like a good practice to use such configuration to load tile + style pack as it requires less code and should work fine.
If loading style pack + tile region separately we need to preserve correct order - first load style pack and only then load tile region otherwise style will be taken from ambient cache and not style pack.

Reply:
Thanks for confirming, but I think we observed some potential different behaviour when building the offline example(cc @julianrex), probably we should verify the it again.